lib/rules/arrow-spacing.js

```diff
@@ -27,16 +27,19 @@ module.exports = {
 
         schema: [
             {
-                type: "object",
-                properties: {
-                    before: {
-                        type: "boolean"
+                oneOf: [
+                    {
+                        enum: ["before", "after", "both", "neither"]
```

Comment:
I'm not sure this is correct. This would allow `before` and `after` as string vales, but they have to be properties of an objects.

Reply:
same settings as well for the yield-star-spacing. I am thinking then to make sure the schema can accommodate the objects proprs but then we need again to make sure they are consistent across the two rules.

Reply:
It’s likely that yield-star-spacing supports the strings for legacy compat; it’s best not to be consistent with it if that’s the case

Reply:
Ah, that actually makes sense. I think is probably the reason why schema for `yield-star-spacing` is like that. We most likely initially had `before` and `after` as string, and then we kept them so we wouldn't cause breaking change. I think they should be removed from `arrow-spacing` as they don't make a whole lot of sense, and I'm not even sure they would work properly.